submissions/maxim-zabolotny/friends-app/app.js

```diff
@@ -0,0 +1,215 @@
+const APP_CONFIG = {
+    url: 'https://randomuser.me/api/?results=60',
+    filterBy: 'name'
+}
+const CSS_CLASSES = {
+    card: 'card',
+    radioButton: 'radio-box',
+    navButton: 'nav-button'
+}
+
+const DOM_NODES = {
+    inputText: document.querySelector('.input-text'),
+    radioButtons: document.querySelectorAll('.radio-box'),
+    form: document.querySelector('.filters'),
+    target: document.querySelector('.frame'),
+    btnWrapper: document.querySelector('.btn-wrapper'),
+    resetBtn: document.querySelector('.filter .reset_button'),
+}
+
+const state = {
+    allUsers : [],
+    filteredUsers: [],
+    sortedUsers: []
+}
+
+window.state = state
+
+const SORT = {
+    all: () => renderCards(state.sortedUsers = [...state.filteredUsers]),
+    female: () => renderCards(state.sortedUsers = state.filteredUsers.filter(user => user.gender.toLowerCase() == 'female')),
+    male: () => renderCards(state.sortedUsers = state.filteredUsers.filter(user => user.gender.toLowerCase() == 'male')),
+    nameAsc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.name.first.toLowerCase() > b.name.first.toLowerCase()) return 1
+        else return -1
+    })),
+    nameDesc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.name.first.toLowerCase() < b.name.first.toLowerCase()) return 1
+        else return -1
+    })),
+    ageAsc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.age > b.age) return 1
+        else return -1
+    })),
+    ageDesc: () => renderCards(state.sortedUsers = state.filteredUsers.sort((a, b) => {
+        if(a.age < b.age) return 1
+        else return -1
+    })),
+}
+
+const capitalized = ({first, last}) => {
+    let str = `${first} ${last}`
+    return str.split(' ').map(s => s.charAt(0).toUpperCase() + s.slice(1)).join(' ');
+}
+
+const formatDate = (date) => {
+    let d = new Date(date)
+    return `${d.getFullYear()}/${d.getMonth()}/${d.getUTCDate()}`
+}
+
+class Card {
+    constructor(props) {
+        this.id = props.id
+        this.photo = props.photo
+        this.name = props.name
+        this.age = props.age
+        this.date = props.date
+        this.email = props.email
+        this.location = props.location
+        this.password = props.password
+        this.gender = props.gender
+        this.cell = props.cell
+    }
+
+    render() {
+        let div = document.createElement('div')
+        div.classList.add('card')
+        div.id=this.id
+        const markup = `<div class="details">
```

Comment:
I suggest putting this markup outside the class. It makes class code hard to read. Maybe a helper that will receive data and return html string? You can prepare the data (uppercase etc.) in your class and then call markup function to get final html.